### PR TITLE
shinyAppTemplate tweaks

### PR DESCRIPTION
**Pull request by wch**
This PR does the following:
* adds the `dryrun` option
* prints out which files are being created
* Errors if a file already exists

### CODE REVIEW

File: R/app_template.R [hunk 1]
```diff
@@ -122,58 +132,79 @@ shinyAppTemplate <- function(path = NULL, examples = "default")
...
+  mkdir <- function(path) {
+    if (!dirExists(path)) {
+      message("Creating ", ensure_trailing_slash(path))
+      if (!dryrun) {
+        dir.create(path, recursive = TRUE)
+      }
+    }
+  }
+
+  copy_file <- function(from, to, is_template = FALSE) {
```
Comment (schloerke) [inline R/app_template.R:144]: Can maybe move the `is_template` logic into the function as it's consistent across the board

Comment (schloerke) [reply R/app_template.R:144]: or make the `is_template` logic a helper function

File: R/app_template.R [hunk 2]
```diff
@@ -109,6 +115,10 @@ shinyAppTemplate <- function(path = NULL, examples = "default")
     return(invisible())
   }
 
+  # =======================================================
```
Comment (schloerke) [inline R/app_template.R:118]: Before this line, can you check for `shinytest` version?

File: R/app_template.R [hunk 3]
```diff
@@ -115,6 +115,17 @@ shinyAppTemplate <- function(path = NULL, examples = "default", dryrun = FALSE)
...
   }
 
+  if ("shinytest" %in% examples) {
+    if (system.file(package = "shinytest") != "" &&
+        packageVersion("shinytest") <= "1.4.0")
+    {
+      message(
+        "The tests/shinytest directory needs shinytest 1.4.0 or later to work properly.\n",
+        "You currently have shinytest ", packageVersion("shinytest"), " installed."
+      )
```
Comment (schloerke) [inline R/app_template.R:125]: Should this be a `stop()`?

Comment (schloerke) [reply R/app_template.R:125]: Wondering if the second line should be displayed if the package is not installed.

Comment (wch) [reply R/app_template.R:125]: I don't think it should be a stop -- we shouldn't force people to change their system for this. But they should know about it.

I agree the second line shouldn't be displayed if shinytest is not installed.

**Pull request merged**